### DISCUSSION TITLE
ModuleNotFoundError: No module named 'Crypto'  & ModuleNotFoundError: No module named 'json'

### DISCUSSION THREAD

**Discussion by hymancheng** [pyinstaller/pyinstaller#5938] [Help] [answered]
my appMain.py import part code as:
`from MainWindow import QMainWindow`

in MainWindow.py , import part code as:
`from Crypto.Cipher import AES`
`import json`

I use Cython to build MainWindow.py to MainWindow.pyd

Then use Pyinstaller to build exe file. But Failed to execute, and tips Error as title.

I have try create hooks-Crypto.py and put it to hooks dir. Then use --additional-hooks-dir hooks in pyinstaller cmd. But it don't works.

My appMain.spec file content as:
`# -*- mode: python ; coding: utf-8 -*-`
`block_cipher = pyi_crypto.PyiBlockCipher(key='12345678')`
`a = Analysis(['appMain.py'],`
`             pathex=['C:\\xxx\\xxxx\\xxx\\'],`
`             binaries=[],`
`             datas=[],`
`            hiddenimports=['MainWindow',],`
`             hookspath=['hooks'],`
`             runtime_hooks=[],`
`             excludes=[],`
`             win_no_prefer_redirects=False,`
`             win_private_assemblies=False,`
`             cipher=block_cipher,`
`             noarchive=False)`
`pyz = PYZ(a.pure, a.zipped_data,cipher=block_cipher)`
`exe = EXE(pyz,`
`          a.scripts,`
`          [],`
`          exclude_binaries=True,`
`          name='appMain',`
`          debug=False,`
`          bootloader_ignore_signals=False,`
`          strip=False,`
`          upx=True,`
`          console=True , version='file_version_info.txt', icon='123.ico')`
`coll = COLLECT(exe,`
`               a.binaries,`
`               a.zipfiles,`
`               a.datas,`
`               strip=False,`
`               upx=True,`
`               upx_exclude=[],`
`               name='appMain')`

1.  I have install pycryptodome by 'pip install pycryptodome' cmd, Version is 3.10.1
2.  ENV is: Win10,  anaconda3(4.9.2), Python(3.8.5)
3.  content of hook-Crypto.py is:
`import os`
`import glob`
`from PyInstaller.compat import EXTENSION_SUFFIXES`
`from PyInstaller.utils.hooks import get_module_file_attribute`
`binaries = []`
`binary_module_names = [`
`'Crypto.Math',`
`'Crypto.Cipher',`
`'Crypto.Util',`
`'Crypto.Hash',`
`'Crypto.Protocol',`
`]`
`try:`
`    for module_name in binary_module_names:`
`        m_dir = os.path.dirname(get_module_file_attribute(module_name))`
`        for ext in EXTENSION_SUFFIXES:`
`            module_bin = glob.glob(os.path.join(m_dir, '_*%s' %ext))`
`            for f in module_bin:`
`                binaries.append((f, module_name.replace('.', os.sep)))`
`except ImportError:`
`pass`

**Comment by rokm** [comment, marked as answer]
> I use Cython to build MainWindow.py to MainWindow.pyd

PyInstaller cannot analyze pyd extensions and therefore cannot automatically pick up the imports made within them. So you'll have to add any imports made within your MainWindow.pyd (and not within any other .py file) to `hiddenimports` - in your case, that would be `json` and `Crypto.Cipher` (or perhaps `Crypto.Cipher.AES`).

**Reply by hymancheng** [reply]
I tried your suggestion and put the imported parts of other files in the ‘appMain.py’ file. 
As a result, pyinstaller and exe runs successfully. 
Thank you.
